src/Encamina.Enmarcha.SemanticKernel/MemoryStoreExtender.cs

```diff
@@ -107,7 +107,7 @@ public virtual async IAsyncEnumerable<string> BatchUpsertMemoriesAsync(string co
 
             if (totalChunks > 0)
             {
-                memoryContent.Metadata.Add(Constants.MetadataTotalChunksCount, totalChunks.ToString());
+                memoryContent.Metadata?.Add(Constants.MetadataTotalChunksCount, totalChunks.ToString());
```

Comment:
In this case, if metadata is null, we should instantiate the variable, so that the value can be added.

Maybe a better alternative would be make Metadata non nullable.